src/pages/experiments/[experimentId]/plots-and-tables/marker-heatmap/index.jsx

```diff
@@ -0,0 +1,307 @@
+import PropTypes from 'prop-types';
+import loadProcessingSettings from '../../../../../redux/actions/experimentSettings/processingConfig/loadProcessingSettings';
+import PlotStyling from '../../../../../components/plots/styling/PlotStyling';
+import { updatePlotConfig, loadPlotConfig } from '../../../../../redux/actions/componentConfig';
+import Header from '../../../../../components/plots/Header';
+import { generateSpec } from '../../../../../utils/plotSpecs/generateHeatmapSpec';
+import { loadGeneExpression, loadMarkerGenes } from '../../../../../redux/actions/genes';
+import { loadCellSets } from '../../../../../redux/actions/cellSets';
+import PlatformError from '../../../../../components/PlatformError';
+import Loader from '../../../../../components/Loader';
+import populateHeatmapData from '../../../../../components/plots/helpers/populateHeatmapData';
+import HeatmapControls from '../../../../../components/plots/styling/heatmap/HeatmapControls';
+
+const { Text } = Typography;
+const { Panel } = Collapse;
+
+const plotUuid = 'markerHeatmapPlotMain';
+const plotType = 'markerHeatmap';
+
+const route = {
+  path: 'marker-heatmap',
+  breadcrumbName: 'Marker-Heatmap',
+};
+const MarkerHeatmap = ({ experimentId }) => {
+  const dispatch = useDispatch();
+  const config = useSelector((state) => state.componentConfig[plotUuid]?.config);
+
+  const { expression: expressionData } = useSelector((state) => state.genes);
+  const { error, loading } = expressionData;
+  const cellSets = useSelector((state) => state.cellSets);
+  const { hierarchy, properties } = cellSets;
+  const selectedGenes = useSelector((state) => state.genes.expression.views[plotUuid]?.data) || [];
+  const [currentSearchedGenes, setCurrentSearchedGenes] = useState([]);
+  const {
+    loading: loadingMarkerGenes,
+    error: errorMarkerGenes,
+  } = useSelector((state) => state.genes.markers);
+  const [vegaSpec, setVegaSpec] = useState();
+  const louvainClustersResolution = useSelector(
+    (state) => state.experimentSettings.processing
+      .configureEmbedding?.clusteringSettings.methodSettings.louvain.resolution,
+  ) || false;
+
+  useEffect(() => {
+    if (!louvainClustersResolution) {
+      dispatch(loadProcessingSettings(experimentId));
+    }
+    dispatch(loadPlotConfig(experimentId, plotUuid, plotType));
+    if (!cellSets.hierarchy.length) {
+      dispatch(loadCellSets(experimentId));
+    }
+  }, []);
+
+  useEffect(() => {
+    if (louvainClustersResolution && config) {
+      dispatch(loadMarkerGenes(experimentId, louvainClustersResolution, plotUuid, config.numGenes));
+    }
+  }, [louvainClustersResolution, hierarchy, config?.numGenes]);
+  const sortGenes = (newGenes) => {
+    const clusters = hierarchy.find((cluster) => cluster.key === config.selectedCellSet).children;
+
+    const getCellIdsForCluster = (clusterId) => properties[clusterId].cellIds;
+
+    const getAverageExpressionForGene = (gene, currentCellIds) => {
```

Comment:
this function is not unit tested either

Reply:
It is tested - its part of the sorting and sorting is tested and working.